pkg/tests/apis/provisioning/helper_test.go

```diff
@@ -84,30 +87,33 @@ func (h *provisioningTestHelper) SyncAndWait(t *testing.T, repo string, options
 
 	name := unstruct.GetName()
 	require.NotEmpty(t, name, "expecting name to be set")
-	h.AwaitJobSuccess(t, t.Context(), name)
+	h.AwaitJobSuccess(t, t.Context(), unstruct)
 }
 
-func (h *provisioningTestHelper) AwaitJobSuccess(t *testing.T, ctx context.Context, jobName string) {
+func (h *provisioningTestHelper) AwaitJobSuccess(t *testing.T, ctx context.Context, job *unstructured.Unstructured) {
 	t.Helper()
+
+	repo := job.GetLabels()[jobs.LabelRepository]
+	require.NotEmpty(t, repo)
 	if !assert.EventuallyWithT(t, func(collect *assert.CollectT) {
-		jobs, err := h.HistoricJobs.Resource.List(ctx, metav1.ListOptions{
-			LabelSelector: jobs.LabelJobOriginalName + "=" + jobName,
```

Comment:
^^^ using name here is likely an error.  It will be duplicated for everything on the same repo with the same action